src/rgw/rgw_sal_rados.cc

```diff
@@ -1302,6 +1302,37 @@ int RadosStore::meta_remove(const DoutPrefixProvider* dpp, std::string& metadata
   return ctl()->meta.mgr->remove(metadata_key, y, dpp);
 }
 
+int RadosStore::list_users(const DoutPrefixProvider *dpp, const std::string& metadata_key,
+                        std::string& marker, int max_entries, void *&handle,
+                        bool* truncated, std::list<std::string>& users)
+{
+  int max = 1000;
+  uint64_t left;
+  uint64_t count = 0;
+
+  if (max_entries > max) {
+      max_entries = max;
+  }
+  int ret = meta_list_keys_init(dpp, metadata_key, marker, &handle);
+  if (ret < 0){
+    return ret;
+  }
+  do {
+    std::list<std::string> keys;
+    left = ((max_entries > 0) ? max_entries - count : max);
```

Comment:
The 'int32_t - uint64_t' is not safe (albeit it won't actually wrap here).
And it will trigger a compiler warning.